sys/kern/tty.c

```diff
@@ -191,16 +353,74 @@ void tty_input(tty_t *tty, uint8_t c) {
     c = '\r';
   }
 
-  if (tty->t_inq.count >= TTY_QUEUE_SIZE) {
-    if (iflag & IMAXBEL) {
-      tty_output(tty, CTRL('g'));
-      tty_notify_out(tty);
+  if (lflag & ICANON) {
+    /* Canonical mode character processing takes place here. */
+    if (CCEQ(cc[VERASE], c)) {
+      /* Erase/backspace */
+      uint8_t erased;
+      if (tty_line_unputc(tty, &erased))
+        tty_erase(tty, erased);
+      goto notify;
```

Comment:
Try to refrain yourself from using `goto`'s in cases where control flow is already complex.

Reply:
Done.